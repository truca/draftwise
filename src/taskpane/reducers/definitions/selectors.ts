import { ReduxState } from "../index";
import { TermsObject } from "./types";

export const getTermsInParagraph =
  (paragraph: string) =>
  (state: ReduxState): string[] => {
    const wordsFromParagraph = paragraph.match(/\w+/g);
    const { terms } = state.definitions;

    return terms.filter((term) => wordsFromParagraph.includes(term));
  };

export const getTermDefinition =
  (term: string) =>
  (state: ReduxState): string | null => {
    const { definitionsHash } = state.definitions;

    return definitionsHash[term];
  };

export const getTerms = (state: ReduxState): string[] => {
  const { terms } = state.definitions;

  return terms;
};

export const getTermsWithDefinitions = (state: ReduxState): TermsObject[] => {
  const terms = getTerms(state);

  return terms.map((term) => {
    const definition = getTermDefinition(term)(state);
    return { term, definition };
  });
};
