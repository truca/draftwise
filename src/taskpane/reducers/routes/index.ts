export enum Routes {
  PARAGRAPH = "PARAGRAPH",
  DOCUMENT = "DOCUMENT",
}

export interface RouteState {
  route: Routes;
}

export enum RouteActionTypes {
  CHANGE_ROUTE = "CHANGE_ROUTE",
}

type ChangeRouteAction = { type: RouteActionTypes.CHANGE_ROUTE; route: Routes };
export type RouteActions = ChangeRouteAction;

export const initialState: RouteState = {
  route: Routes.DOCUMENT,
};

const routes = (state: RouteState = initialState, action?: RouteActions) => {
  switch (action?.type) {
    case RouteActionTypes.CHANGE_ROUTE:
      return { ...state, route: action.route };
    default:
      return state;
  }
};

export default routes;
