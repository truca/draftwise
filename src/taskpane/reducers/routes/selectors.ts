import { Routes } from "./index";
import { ReduxState } from "../index";

export const routeSelector = (state: ReduxState): Routes => {
  return state.routes.route;
};
