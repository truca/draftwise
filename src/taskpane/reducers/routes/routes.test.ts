import routes, { initialState, Routes, RouteActionTypes } from "./index";
import { routeSelector } from "./selectors";

describe("routes", () => {
  describe("reducer", () => {
    it("initial state", () => {
      const state = routes();
      expect(state).toMatchSnapshot();
    });
    it("change route", () => {
      const state = routes(initialState, { type: RouteActionTypes.CHANGE_ROUTE, route: Routes.PARAGRAPH });
      expect(state).toMatchSnapshot();
    });
  });

  describe("selectors", () => {
    it("routeSelector", () => {
      const result = routeSelector({
        routes: {
          route: Routes.DOCUMENT,
        },
      } as any);
      expect(result).toMatchSnapshot();
    });
  });
});
