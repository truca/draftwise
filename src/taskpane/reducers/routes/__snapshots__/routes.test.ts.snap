// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes reducer change route 1`] = `
{
  "route": "PARAGRAPH",
}
`;

exports[`routes reducer initial state 1`] = `
{
  "route": "DOCUMENT",
}
`;

exports[`routes reducer initial state 2`] = `
{
  "route": "PARAGRAPH",
}
`;

exports[`routes selectors routeSelector 1`] = `"DOCUMENT"`;
